packages/doenetml-worker-rust/lib-doenetml-core/src/core/general_state_var/boolean_state_var.rs

```diff
@@ -158,15 +158,13 @@ impl StateVarUpdater<bool, BooleanRequiredData> for BooleanStateVar {
             return StateVarCalcResult::Calculated(false);
         } else if data.base.len() > 1 {
             // Have multiple string variables. Concatenate the string values into a single string
-            // TODO: can we do this without cloning?
-            let value: String = data
-                .base
-                .iter()
-                .map(|v| match v {
-                    BooleanOrString::Boolean(boolean_val) => boolean_val.get().to_string(),
-                    BooleanOrString::String(string_value) => string_value.get().to_string(),
-                })
-                .collect();
+
+            let mut value = String::new();
+            value.extend(data.base.iter().map(|v| match v {
+                BooleanOrString::Boolean(boolean_val) => boolean_val.get().to_string(),
```

Comment:
Does it work without the `to_string` now? That was supposed to be the whole point of switching to this method (because `extend` can work on references).

Reply:
No, it doesn't currently work if I remove the `.to_string()` on the strings. In this case with the `Boolean`, it gets mad about incompatible match arm types. In `string_state_var.rs`, it gets made because the output of `.get()` is currently typed as `impl Deref<Target = String> + '_`, which doesn't implement `Expand`.